python_modules/dagster/dagster/_core/definitions/metadata/__init__.py

```diff
@@ -441,6 +441,41 @@ def validate_table(context, df):
         check.inst_param(asset_key, "asset_key", AssetKey)
         return DagsterAssetMetadataValue(asset_key)
 
+    @public
+    @staticmethod
+    def job(
+        job_name: str,
+        location_name: Optional[str] = None,
+        repository_name: Optional[str] = None,
+    ) -> "DagsterJobMetadataValue":
+        """Static constructor for a metadata value referencing a Dagster job, by name.
+
+        For example:
+
+        .. code-block:: python
+
+            @op
+            def emit_metadata(context, df):
+                yield AssetMaterialization(
+                    asset_key="my_dataset"
+                    metadata={
+                        "Producing job": MetadataValue.job('my_other_job'),
+                    },
+                )
+
+        Args:
+            job_name (str): The name of the job.
+            location_name (Optional[str]): The location name of the job. If not provided, the job is
+                assumed to be in the same location as this object.
+            repository_name (Optional[str]): The repository name of the job. If not provided, the
+                job is assumed to be in the same repository as this object.
+        """
+        return DagsterJobMetadataValue(
+            job_name=check.str_param(job_name, "job_name"),
+            location_name=check.opt_str_param(location_name, "location_name"),
```

Comment:
It looks like this is optional here, but required in `DagsterJobMetadataValue`?

Reply:
Good catch